jepsen/src/jepsen/nemesis/combined.clj

```diff
@@ -359,6 +359,60 @@
                            :fs    #{:strobe-clock}
                            :color "#A0E9E3"}}}))
 
+(defn file-corruption-package
+  "A nemesis and generator package that corrupts files.
+   
+   Opts:
+   ```clj
+   {:file-corruption
+    {:targets     [...] ; A collection of node specs, e.g. [:one, [\"n1\", \"n2\"], :all]
+     :corruptions [     ; A collection of file corruptions, e.g.:
+      {:bitflip \"/path/to/file\"
+       :probability 1e-3},
+      {:bitflip \"path/to/dir\"
+       :probability [1e-3 1e-4 1e-5]},
+      {:truncate \"path/to/file/or/dir\"
+       :drop (range 1 1025)}]}}
+   ```
+   
+   If a directory is given to `:bitflip` or `:truncate`,
+   a new random file is selected from that directory on each target node for each operation.
+
+   If `:probability` or `:drop` is a collection, a new `rand-nth` value is chosen for each operation.
+   
+   See [[jepsen.nemesis/bitflip]] and [[jepsen.nemesis/truncate-file]].
+
+   Additional options as for [[nemesis-package]]."
+  [{:keys [faults db file-corruption interval nodes] :as _opts}]
+  (let [needed?     (:file-corruption faults)
+        targets     (:targets     file-corruption (node-specs db))
+        corruptions (:corruptions file-corruption)
+        gen (->> (fn gen [_test _context]
+                   (let [targets     (->> (rand-nth targets)
+                                          (db-nodes {:nodes nodes} db))
+                         corruption  (rand-nth corruptions)
+                         {:keys [bitflip truncate probability drop]} corruption
+                         probability (->> probability (util/coll) (rand-nth))
```

Comment:
Yeah this rand-nth is gonna do linear traversal over the ranges. Honestly we do enough probabilistic selection stuff that it might be worth having a standard function in jepsen.util for drawing a random value given a map like {:distribution :uniform, :min 1, :max 50}. Then we can just say that probability and drop take distribution maps.

Reply:
> might be worth having a standard function in jepsen.util for drawing a random value given a map like {:distribution :uniform, :min 1, :max 50}

That's very expressive.
Will encourage use of more than a uniform distribution too.

Will commit and summarize in the main conversation thread.